### PR TITLE
MAT-5055: Display only one error if there's no population criteria

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-5055](https://jira.cms.gov/browse/MAT-5055)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/routes/qdm/TestCaseRoutes.tsx [hunk 1]
```diff
@@ -60,10 +60,17 @@ const TestCaseRoutes = () => {
           })
           .catch((err) => {
             setContextFailure(true);
-            setErrors((prevState) => [
-              ...prevState,
-              "An error occurred, please try again. If the error persists, please contact the help desk",
-            ]);
+            setErrors((prevState) =>
+              prevState.length === 1 &&
```
Comment (RohitKandimalla) [inline src/components/routes/qdm/TestCaseRoutes.tsx]: Looks like when measure.groups are empty then `convertToCqmMeasure` will throw an exception, if that is the case, Do you think we could skip the `convertToCqmMeasure` altogether when Population Criteria is not set ?

convertToCqmMeasure requires relevant DataElements, and Relevant DataElements requires Population Criteria

Comment (RohitKandimalla) [reply src/components/routes/qdm/TestCaseRoutes.tsx]: Also we are only removing duplicate error exception only when prevState[0] has that particular error about PC, but in the following screenshot, its the 2nd error message and not first. 

image

File: src/components/routes/qiCore/TestCaseRoutes.tsx [hunk 2]
```diff
@@ -52,47 +53,37 @@ const TestCaseRoutes = () => {
...
-      });
+
+      if (
+        measure?.errors?.includes(
+          MeasureErrorType.MISMATCH_CQL_POPULATION_RETURN_TYPES
+        )
+      ) {
+        localErrors.push(CQL_RETURN_TYPES_MISMATCH_ERROR);
+      }
+      localErrors.filter((s) => s !== CQL_RETURN_TYPES_MISMATCH_ERROR);
```
Comment (RohitKandimalla) [inline src/components/routes/qiCore/TestCaseRoutes.tsx]: This has to be in else block.

**Review by RohitKandimalla — CHANGES_REQUESTED** [review]

File: src/components/routes/qiCore/TestCaseRoutes.tsx (hunk 2 above)
Comment (ethankaplan) [reply src/components/routes/qiCore/TestCaseRoutes.tsx]: Yep, forgot to commit/push it

Comment (RohitKandimalla) [reply src/components/routes/qiCore/TestCaseRoutes.tsx]: Filter will not mutate the object, instead it returns a new array, so we need to assign the result it back to localErrors array

Comment (RohitKandimalla) [reply src/components/routes/qiCore/TestCaseRoutes.tsx]: Atleast that is what my understanding is, I was hoping there are unit test cases to prove me wrong.

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**